### PR TITLE
Add promql executor for unit tests

### PR DESCRIPTION
**Pull request by Argh4k**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

This new executor uses prometheus query engine with lazy loader backend, allowing users to load time series from file. Thanks to that we are able to write unit tests for measurements which run queries against executor that can parse and execute queries in the same manner as prometheus does. Therefore, we will be able to detect issues with our queries in measurement. Currently in unit tests, we use fakeExecutors which always return the same results for test case even after the query in measurement is changed.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
The test data for all cases begins with multiple 0s, becasue first 5 minutes of series are ignored in api_responsiveness_prometheus.go (https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus.go#L186)

Also data after first 5 minutes must not start with 0:
```
If b has 0 observations, NaN is returned. If b contains fewer than two buckets, NaN is returned. For φ < 0, -Inf is returned. For φ > 1, +Inf is returned.
```
(https://prometheus.io/docs/prometheus/latest/querying/functions/#histogram_quantile)

**Does this PR introduce a user-facing change?**:
no

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: clusterloader2/pkg/measurement/common/slos/testdata/rules.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+groups:
```
Comment (mborsz) [inline clusterloader2/pkg/measurement/common/slos/testdata/rules.yml:1]: Is it a copy of https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/prometheus/manifests/prometheus-rules.yaml?

If so, can we reuse that file? I think it does make sense as the rules there should be technically considered a part of measurement (it's bad that they are in some separate file, but it's a different story).
If not, can we at last add some comment to that file to keep this file in sync?

Comment (Argh4k) [reply clusterloader2/pkg/measurement/common/slos/testdata/rules.yml:1]: Yes, these are the same rules as in prometheus-rules.yaml. We cannot reuse it 1:1 as prometheus-rules.yaml is actually k8s resource definition and rules are inside specification, but it should not be hard to write some logic that would extract them.

**Comment by mborsz** [comment]
Looks good overal. Few nits.

File: clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus_test.go [hunk 2]
```diff
@@ -86,6 +90,53 @@ type fakeQueryExecutor struct {
...
+		Quantile string `yaml:"quantile"`
+	} `yaml:"labels"`
+}
+
+type group struct {
+	Name  string `yaml:"name"`
+	Rules []rule `yaml:"rules"`
+}
+
+type prometheusRuleManifest struct {
```
[the remark this answers is not in the document]
Comment (Argh4k) [reply clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus_test.go:106]: I don't think it's worth the time to add this as dependency. prometheus-operator package depends on specific version of prometheus/prometheus package (commit based) and also on k8s.io packages that we do not want to upgrade. Finding proper version of prometheus-operator package that would not interfere with our current dependencies seems like an overkill to just import one struct. I will just add a comment.

[the remark this answers is not in the document]
Comment (mborsz) [reply clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus_test.go:108]: Makes sense, Thanks!

**Pull request merged**